### PR TITLE
Added the ability to change properties of UiActions

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

File: src/framework/ui/iuiactionsmodule.h [hunk 1]
```diff
@@ -35,6 +35,14 @@ class IUiActionsModule
     virtual ~IUiActionsModule() = default;
 
     virtual const UiActionList& actionsList() const = 0;
+    virtual async::Channel<UiAction> actionChanged() const
```
Comment (RomanPudashkin) [inline src/framework/ui/iuiactionsmodule.h:38]: Perhaps we should change this channel a bit to be able to send multiple actions at once:

async::Channel< **UiActionList** > action**s**Changed() const

That would prevent potential notification spam

Comment (igorkorsukov) [reply src/framework/ui/iuiactionsmodule.h]: changed to list

File: src/framework/ui/iuiactionsregister.h [hunk 2]
```diff
@@ -38,10 +38,13 @@ class IUiActionsRegister : MODULE_EXPORT_INTERFACE
 
     virtual void reg(const IUiActionsModulePtr& actions) = 0;
 
-    virtual const UiAction& action(const muse::actions::ActionCode& code) const = 0;
-    virtual const std::vector<UiAction> getActions() const = 0;
-    virtual UiActionState actionState(const muse::actions::ActionCode& code) const = 0;
-    virtual async::Channel<muse::actions::ActionCodeList> actionStateChanged() const = 0;
+    virtual const std::vector<UiAction> actionList() const = 0;
```
Comment (saintmatthieu) [inline src/framework/ui/iuiactionsregister.h:41]: Why a `const` return value if it's not a reference?

File: src/framework/ui/internal/uiactionsregister.h [hunk 3]
```diff
@@ -69,6 +73,8 @@ class UiActionsRegister : public IUiActionsRegister, public Injectable, public a
     void updateShortcuts(const actions::ActionCodeList& codes);
     void updateShortcutsAll();
 
+    void updateActions(const UiActionList& actions);
```
Comment (saintmatthieu) [inline src/framework/ui/internal/uiactionsregister.h:76]: Not `override` ? Aren't clients supposed to interact with abstract `IUiActionsRegister` ?

**Review by saintmatthieu — COMMENTED** [review]
I'm surprised that the method to modify the actions isn't part of the abstract interface.
Besides that LGTM.

**Comment by igorkorsukov** [comment]
> I'm surprised that the method to modify the actions isn't part of the abstract interface. Besides that LGTM.

There are two reasons for this: 
1. We had no need to change the properties of actions for the user.
2. It is made so that actions are managed and owned only by the module to which they belong, which processes them. Accordingly, there is no need to provide the possibility for someone from outside to change some action in some module.

File: src/framework/ui/internal/uiactionsregister.h (hunk 3 above)
Comment (igorkorsukov) [reply src/framework/ui/internal/uiactionsregister.h:76]: not override
the reasons are described here (number two)
https://github.com/musescore/MuseScore/pull/24834#issuecomment-2363578645

File: src/framework/ui/iuiactionsregister.h (hunk 2 above)
Comment (igorkorsukov) [reply src/framework/ui/iuiactionsregister.h]: removed const 
(I don't know why the author added const, probably wanted to somehow semantically show that there is no need to change this list...
I just renamed the method to match our style)

**Comment by saintmatthieu** [comment]
Can't find the "Approve" button, but I approve :) Thanks !

**Comment by shoogle** [comment]
> Can't find the "Approve" button

[Files changed tab](https://github.com/musescore/MuseScore/pull/24834/files) > Review changes.

However, the 'approve' option has disappeared now that the PR is merged.

**Pull request merged**